backend-worker/ucm_valuation.py

```diff
@@ -0,0 +1,494 @@
+    temp_dir.mkdir(exist_ok=True)
+
+    # Calculate Heating Degree Days raster
+    logger.debug(f"Calculating Heating Degree Days")
+    hdd_tif = Path(args["workspace_dir"]) / f"hdd{file_suffix}.tif"
+    hdd_calculation(args["air_temp_tif"], hdd_tif)
+
+    # Calculate energy use raster (kWh) based on Heating Degree Days
+    hdd_kwh_tif = Path(args["workspace_dir"]) / f"hdd_kwh{file_suffix}.tif"
+    grouped_scalar_calculation(
+        hdd_tif,
+        args["lulc_tif"],
+        hdd_kwh_tif,
+        dd_energy_df["lucode"].to_list(),
+        dd_energy_df["kwh_per_hdd"].to_list(),
+        temp_dir,
+    )
+
+    # Calculate energy use raster ($) based on Heating Degree Days
+    hdd_cost_tif = Path(args["workspace_dir"]) / f"hdd_cost{file_suffix}.tif"
+    grouped_scalar_calculation(
+        hdd_kwh_tif,
+        args["lulc_tif"],
+        hdd_cost_tif,
+        dd_energy_df["lucode"].to_list(),
+        dd_energy_df["cost_per_kwh"].to_list(),
+        temp_dir,
+    )
+
+    # Calculate Cooling Degree Days raster
+    logger.debug(f"Calculating Cooling Degree Days")
+    cdd_tif = Path(args["workspace_dir"]) / f"cdd{file_suffix}.tif"
+    cdd_calculation(args["air_temp_tif"], cdd_tif)
+
+    # Calculate energy use raster (kWh) based on Cooling Degree Days
+    cdd_kwh_tif = Path(args["workspace_dir"]) / f"cdd_kwh{file_suffix}.tif"
+    grouped_scalar_calculation(
+        cdd_tif,
+        args["lulc_tif"],
+        cdd_kwh_tif,
+        dd_energy_df["lucode"].to_list(),
+        dd_energy_df["kwh_per_cdd"].to_list(),
+        temp_dir,
+    )
+
+    # Calculate energy use raster ($) based on Cooling Degree Days
+    cdd_cost_tif = Path(args["workspace_dir"]) / f"cdd_cost{file_suffix}.tif"
+    grouped_scalar_calculation(
+        cdd_kwh_tif,
+        args["lulc_tif"],
+        cdd_cost_tif,
+        dd_energy_df["lucode"].to_list(),
+        dd_energy_df["cost_per_kwh"].to_list(),
+        temp_dir,
+    )
+
+    # Calculate Mortality Risk
+    logger.debug(f"Calculating Relative Mortality Risk")
+
+    mortality_risk_df = pd.read_csv(
+        args["mortality_risk_path"], encoding="unicode_escape"
+    )
+
+    # Test for correct csv headers
+    for header in _EXPECTED_MORTALITY_HEADERS:
+        if header not in mortality_risk_df.columns:
+            raise ValueError(
+                f"Expected a header in biophysical table that matched the pattern '{header}' but was unable to find "
+                f"one. Here are all the headers from {args['mortality_risk_path']}: {list(mortality_risk_df.columns)}"
+            )
+
+    # Test if selected city is in the Guo et al dataset
+    city_name_global = f"{args['city'].split(',')[0]}, USA"
+    try:
+        if city_name_global not in mortality_risk_df["city"]:
+            raise IndexError("GuoStudy")
+    except IndexError:
+        logger.error(
+            f"'{city_name_global}' not in Guo et al. 2014 Mortality Risk study"
+        )
+    else:
```

Comment:
I also don't use try/except much at all, I was just experimenting with learning the form when I wrote this since I had seen it around and figured it would be worth a 'try'. I prefer if/else, so I will re-write with your suggestion!